### ISSUE TITLE
Break up the largest repos into batches

### ISSUE THREAD

**Issue by dabutvin** [dabutvin/ImgBot#382]
When we encounter a repo with, for example, 10,000 images in it, we struggle to work through the standard compression routine and in some cases, we timeout eventually.

I am proposing we split the workflow up based on image count. This is what I'm thinking right now:

- query images as usual, but add a sort to the found images
- if there are more than 1,000 images found, only consider the first 1,000
- if we complete the compression routine and there are none to optimize, queue a new message with a new property `page=2`
- if we have a page property when the message is incoming, skip to the 1,001-,2000 using the same query.
- if we complete the current page without finding any images, queue the next page in a new message

All these 'pages' of images would use the same `imgbot` branch and we would not do the next page until the current page was integrated and the branch deleted.

These pages would stay segmented indefinitely, even as images on multiple pages changed, we would open separate PRs to optimize them.

**Comment by guntbert** [comment]
I don't have a better idea but everything that stays "indefinitely" raises concerns of pending unmanageability.

**Comment by dabutvin** [comment]
hey @guntbert ! 

definitely a good flag to raise.
Ideally we would collapse the image changes across multiple groups into a single PR, but not sure the best way to pull that off today.

**Comment by dabutvin** [comment]
I think I have a solution to this figured out.

Since we have access to the set of images that are added/modified when we trigger for ongoing optimization, we can pull this set of images onto the first page so that they don't get broken up across multiple prs.

**Comment by dabutvin** [comment]
this got a lot more complicated than I expected, but I did stumble on a new idea - adding multiple commits into a single PR

**Comment by dabutvin** [comment]
so after running around in circles on this, I believe the long processing of the large repos was never the issue. I believe that the optimization of an individual image created an infinite loop that never returned and we got stuck in the optimize loop.

I believe this because, when monitoring an optimization run that was previously never completed, I noticed we ran through all the images much faster than I expected, but still never completed. The CPU stayed spiked like it was doing work, but never finishing.

I opened #390 to add a 10 second timeout to each image process. Pending a full test to validate.
